Helper/PaymentFailureReasonHelper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Copyright © TrueLayer Ltd. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace TrueLayer\Connect\Helper;
+
+use Magento\Framework\Phrase;
+
+class PaymentFailureReasonHelper
+{
+    public static function getHumanReadableLabel(string $reason = null): Phrase
+    {
+        switch ($reason) {
+            case 'cancelled':
+                return __('You cancelled the payment.');
+            case 'expired':
+                return __('Payment has expired.');
+            case 'not_authorized':
+                return __('Payment was not authorized.');
+            case 'provider_error':
+                return __('Your provider has encountered an error.');
```

Comment:
If this is displayed to the end user, I'm not sure they will know who we're referring to by `provider`.